### PR TITLE
#3 Added ICouchEventLog with log4net based implementation

### PR DESCRIPTION
**Pull request by pbakshy**

### CODE REVIEW

File: Edge10.CouchDb.Client/CouchApi.cs [hunk 1]
```diff
@@ -42,12 +44,15 @@ public CouchApi(ICouchDbConnectionStringBuilder connectionString) : this(connect
 		/// </summary>
 		/// <param name="connectionString">The connection string.</param>
 		/// <param name="httpClient">The HTTP client facade.</param>
-		internal CouchApi(ICouchDbConnectionStringBuilder connectionString, IHttpClientFacade httpClient)
+		/// <param name="eventLog">The couch event log.</param>
+		internal CouchApi(ICouchDbConnectionStringBuilder connectionString, IHttpClientFacade httpClient, ICouchEventLog eventLog)
```
Comment (stevegreatrex) [inline Edge10.CouchDb.Client/CouchApi.cs:48]: Do we need a separate internal constructor?  Could we not use the constructor above with a default parameter?

File: Edge10.CouchDb.Client/CouchTimelineEventLog.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		/// <param name="action">The action.</param>
+		/// <returns></returns>
+		public IDisposable LogDocumentEvent(string documentId, string action)
+		{
+			return Time(action, documentId);
+		}
+
+		private IDisposable Time(string eventName, string eventSubText)
+		{
+			var start = DateTime.UtcNow;
```
Comment (stevegreatrex) [inline Edge10.CouchDb.Client/CouchTimelineEventLog.cs]: I know this is how we did timings in the old version but I think we can get more accuracy using `System.Diagnostics.Stopwatch`

**Review by stevegreatrex — APPROVED** [review]
LGTM.  2 possible changes/improvements

File: Edge10.CouchDb.Client/CouchApi.cs (hunk 1 above)
Comment (pbakshy) [reply Edge10.CouchDb.Client/CouchApi.cs:48]: Yeah, the idea to wrap HttpClient for testing but keep it internal so I use this constructor for testing only

**Review by stevegreatrex — APPROVED** [review]

**Pull request merged**